File: server/src/main/com/broll/gainea/server/core/goals/impl/all/G_OccupyShips.java
package com.broll.gainea.server.core.goals.impl.all;

import com.broll.gainea.server.core.bot.strategy.GoalStrategy;
import com.broll.gainea.server.core.goals.CustomOccupyGoal;
import com.broll.gainea.server.core.goals.GoalDifficulty;
import com.broll.gainea.server.core.map.Ship;

import java.util.HashSet;

public class G_OccupyShips extends CustomOccupyGoal {
    private final int ships;

    public G_OccupyShips() {
        this(GoalDifficulty.EASY, 5);
    }

    public G_OccupyShips(GoalDifficulty difficulty, int ships) {
        super(difficulty, "Erobere " + ships + " Schiffe");
        this.ships = ships;
        setProgressionGoal(ships);
    }

    @Override
    public void check() {
        long count = player.getControlledLocations().stream().filter(it -> it instanceof Ship).count();
        updateProgression((int) count);
        if (count >= ships) {
            success();
        }
    }

    @Override
    public void botStrategy(GoalStrategy strategy) {
        strategy.setRequiredUnits(ships);
        strategy.updateTargets(new HashSet<>(game.getMap().getAllShips()));
    }
}

File: server/src/main/com/broll/gainea/server/core/goals/impl/all/G_OccupyShips.kt
package com.broll.gainea.server.core.goals.impl.all

import com.broll.gainea.server.core.bot.strategy.GoalStrategy
import com.broll.gainea.server.core.goals.CustomOccupyGoal
import com.broll.gainea.server.core.goals.GoalDifficulty
import com.broll.gainea.server.core.map.Ship

open class G_OccupyShips(difficulty: GoalDifficulty = GoalDifficulty.EASY, private val ships: Int = 5) :
        CustomOccupyGoal(difficulty, "Erobere $ships Schiffe") {
    init {
        setProgressionGoal(ships)
    }

    override fun check() {
        val count = player.controlledLocations.count { it is Ship }
        updateProgression(count)
        if (count >= ships) {
            success()
        }
    }

    override fun botStrategy(strategy: GoalStrategy) {
        strategy.setRequiredUnits(ships)
        strategy.updateTargets(game.map.allShips.toSet())
    }
}